src/python/tests/core/bot/fuzzers/libFuzzer/libfuzzer_launcher_test.py

```diff
@@ -1804,6 +1817,348 @@ def test_fuzz_with_mutations_using_ml_rnn(self, mock_execute, *_):
+                  0,
+              u'slowest_unit_time_sec':
+                  0,
+              u'startup_crash_count':
+                  0,
+              u'strategy_dataflow_tracing':
+                  0,
+              u'strategy_corpus_mutations_radamsa':
+                  0,
+              u'strategy_corpus_mutations_ml_rnn':
+                  0,
+              u'strategy_corpus_subset':
+                  0,
+              u'strategy_fork':
+                  1,
+              u'strategy_mutator_plugin':
+                  0,
+              u'strategy_random_max_len':
+                  0,
+              u'strategy_recommended_dict':
+                  0,
+              u'strategy_value_profile':
+                  0,
+              u'timeout_count':
+                  0,
+              u'timeout_limit':
+                  25,
+              'timestamp':
+                  1337.0
+          })
+
+      # Output printed.
+      self.assertIn(self.no_crash_output, mock_stdout.getvalue())
+      expected_command = (
+          'Command: /fake/build_dir/fake_fuzzer '
+          '-max_len=80 -rss_limit_mb=2048 -timeout=25 '
+          '-collect_data_flow=/fake/dfsan_build/fake_fuzzer -fork=1 '
+          '-artifact_prefix=/fake/ -max_total_time=2650 -print_final_stats=1 '
+          '/fake/inputs-disk/temp-1337/new /fake/corpus_dir')
+      self.assertIn(expected_command, mock_stdout.getvalue())
+      self.assertIn('Bot: test-bot', mock_stdout.getvalue())
+      self.assertIn('Time ran:', mock_stdout.getvalue())
+
+  @mock.patch('bot.fuzzers.libFuzzer.launcher.add_recommended_dictionary',
+              lambda x, y, z: False)
+  @mock.patch('multiprocessing.cpu_count', return_value=1)
+  @mock.patch('sys.stdout', new_callable=test_utils.MockStdout)
+  @mock.patch('system.minijail.tempfile.NamedTemporaryFile')
+  def test_fuzz_with_dataflow_tracing_minijail(self, mock_tempfile, *_):
+    """Tests fuzzing with dataflow tracing inside minijail."""
+    self.mock.do_dataflow_tracing.return_value = True
+    os.environ['USE_MINIJAIL'] = 'True'
+
+    mock_tempfile.return_value.__enter__.return_value.name = '/tmppath'
+    mock_tempfile.return_value.name = '/tmpfile'
+
+    self.fs.CreateDirectory('/fake/corpus_dir')
+    self.fs.CreateFile('/fake/testcase_basic')
+    self.fs.CreateDirectory('/fake/dfsan_build')
+    self.fs.CreateFile('/fake/dfsan_build/fake_fuzzer')
+
+    os.environ['FUZZ_CORPUS_DIR'] = '/fake/corpus_dir'
+    os.environ['DATAFLOW_BUILD_DIR'] = '/fake/dfsan_build'
+
+    new_units_added = 11
+    with mock.patch(
+        'subprocess.Popen',
+        create_mock_popen(
+            self.no_crash_output, '/fake/inputs-disk/temp-1337/new',
+            '/fake/main_corpus_dir', new_units_added)) as mock_popen:
+      launcher.main([
+          'launcher.py',
+          '/fake/testcase_basic',
+          'fake_fuzzer',
+          '-max_len=80',
+      ])
+
+      self.assertEqual(mock_popen.commands, [[
```

Comment:
2039-2069 seems to be duplicated across minijail tests, can you please find a way to abstract this assertion to avoid this?

Reply:
Added to #594 for potential test refactoring.